akka-diagnostics/src/main/scala/akka/diagnostics/StarvationDetector.scala

```diff
@@ -232,7 +233,10 @@ object StarvationDetector {
                   s"Stack traces:\n$stacks"
                 }
               case Failure(ex) =>
-                s"[Could not get thread info because ${ex.toString}]"
+                ex match {
+                  case _: InaccessibleObjectException => throw ex //stopping Starvation Detector
```

Comment:
This is a problem for validation on JDK 8 because this exception doesn't exist. I can think in two solutions:
1. catching the exception by searching for certain message in it and build my own exception out of it.
2. forget about giving a warning and let the next [catch](https://github.com/akka/akka-diagnostics/pull/20/files#diff-a670e72863a7f1e8a29d2d5ff98b5a0e39aad9101eae93691e74c95622f0d00eL264) log an error and stop the Starvation detector.

Reply:
The problem with 1. is that is a bit messy. 
The problem with 2. is that it might be confusing for the user to read `Starvation detector failed and terminated. This is likely a bug. Please report to Lightbend support.`

Reply:
maybe this could work:
```
case exc: RuntimeException if exc.getClass.getName == "java.lang.reflect.InaccessibleObjectException" =>
  // InaccessibleObjectException doesn't exist in JDK 8
```

Reply:
well, maybe what I wrote is also fine. You'll be the judge.